subscription-fee/src/subtract_payments.rs

```diff
@@ -52,16 +49,15 @@ pub trait SubtractPaymentsModule:
         let caller = self.blockchain().get_caller();
         let service_id = self.service_id().get_id_non_zero(&caller);
 
-        let service_info = self.service_info(service_id).get().get(service_index);
-        let opt_cost = self.get_user_cost(user_id, &service_info);
-        if opt_cost.is_none() {
+        let opt_user_address = self.user_id().get_address(user_id);
+        if opt_user_address.is_none() {
             return MyVeryOwnScResult::Err(());
         }
 
-        let cost = unsafe { opt_cost.unwrap_unchecked() };
-        let subtract_result = match service_info.payment_type.opt_specific_token {
-            Some(token_id) => self.subtract_specific_token(user_id, token_id, cost),
-            None => self.subtract_any_token(user_id, cost),
+        let service_info = self.service_info(service_id).get().get(service_index);
+        let subtract_result = match service_info.opt_payment_token {
```

Comment:
We have to check whether the user have subscribed for this service, and when was the last time this service made a subtract payment from this user. To check the periodicity.

Reply:
Already done in `subscriber/daily_operation.rs`, line 156.